File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hashMap = {}

        for number in nums:
            if number not in hashMap:
                hashMap[number] = 1
            else:
                hashMap[number] += 1

        max_key = max(hashMap, key=hashMap.get)    
        return max_key

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int result = 0, count = 0;

        for (int num: nums) {
            if (count == 0) {
                result = num;
            }

            count += (num == result) ? 1 : -1;
        }

        return result;
    }
}